server/neptune/workflows/internal/deploy/revision/queue/worker.go

```diff
@@ -105,10 +133,35 @@ func (w *Worker) Work(ctx workflow.Context) {
 	}
 }
 
+func (w *Worker) isValidRevision(ctx workflow.Context, msg terraform.DeploymentInfo, resp activities.CompareCommitResponse) bool {
+	switch resp.CommitComparison {
+	case activities.DirectionBehind:
+		logger.Info(ctx, fmt.Sprintf("Deployed Revision: %s is ahead of the Deploy Request Revision: %s, moving to next one", w.LatestDeployment.Revision, msg.Revision))
+		w.updateCheckRun(ctx, msg, github.CheckRunFailure, DirectionBehindSummary)
+		return false
+
+	case activities.DirectionIdentical:
+		logger.Info(ctx, fmt.Sprintf("Deployed Revision: %s is identical to the Deploy Request Revision: %s", w.LatestDeployment.Revision, msg.Revision))
+		return true
+
+	// TODO: Handle Force Applies
+	case activities.DirectionDiverged:
+		logger.Info(ctx, fmt.Sprintf("Deployed Revision: %s is divergent from the Deploy Request Revision: %s.", w.LatestDeployment.Revision, msg.Revision))
```

Comment:
why are you attaching LatestDeployment to the struct? This doesn't need to be an exported member at all does it? Can't it just be local to the work function?